include/exadg/acoustic_conservation_equations/spatial_discretization/operators/weak_boundary_conditions.h

```diff
@@ -65,27 +66,45 @@ template<int dim, typename Number>
 inline DEAL_II_ALWAYS_INLINE //
   dealii::Tensor<1, dim, dealii::VectorizedArray<Number>>
   calculate_exterior_value_velocity(unsigned int const                       q,
-                                    FaceIntegrator<dim, dim, Number> const & integrator_m,
+                                    FaceIntegrator<dim, dim, Number> const & velocity_m,
+                                    FaceIntegrator<dim, 1, Number> const &   pressure_m,
+                                    double const                             c,
                                     BoundaryType const &                     boundary_type,
                                     dealii::types::boundary_id const         boundary_id,
                                     BoundaryDescriptor<dim> const &          boundary_descriptor,
                                     Number const                             time)
 {
-  (void)boundary_id;
-  (void)boundary_descriptor;
-  (void)time;
-
   if(boundary_type == BoundaryType::VelocityDirichlet)
   {
     auto const g = FunctionEvaluator<1, dim, Number>::value(
       *boundary_descriptor.velocity_dbc.find(boundary_id)->second,
-      integrator_m.quadrature_point(q),
+      velocity_m.quadrature_point(q),
       time);
-    return -integrator_m.get_value(q) + Number{2.0} * g;
+    return -velocity_m.get_value(q) + Number{2.0} * g;
   }
   else if(boundary_type == BoundaryType::PressureDirichlet)
   {
-    return integrator_m.get_value(q);
+    return velocity_m.get_value(q);
+  }
+  else if(boundary_type == BoundaryType::Admittance)
+  {
+    auto const Y = FunctionEvaluator<0, dim, Number>::value(
+      *boundary_descriptor.admittance_bc.find(boundary_id)->second,
+      velocity_m.quadrature_point(q),
+      time);
+
+    auto const n      = velocity_m.normal_vector(q);
+    auto const rho_um = velocity_m.get_value(q);
+    auto const pm     = pressure_m.get_value(q);
+
+    auto const rho_um_normal     = (rho_um * n) * n;
+    auto const rho_um_tangential = rho_um - rho_um_normal;
+
+    // normal share of velocty
+    auto const rho_u_n = (Number{2.0} * Y * pm / c) * n - rho_um_normal;
+
+    // tangential share of velocity stays the same
```

Comment:
For clarity, I would add here "(i.e. rho_up_tangential = rho_um_tangential)".

Reply:
[af66293](https://github.com/exadg/exadg/pull/629/commits/af6629314c29799f74213d1444ae367f1ac49c2f)